test/gtest/graphapi_opgraph.cpp

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <gtest/gtest.h>
+
+#include <miopen/miopen.h>
+
+#include <miopen/graphapi/opgraph.h>
+
+namespace gr = miopen::graphapi;
+
+template <bool IsVirtual>
+gr::Tensor makeDummyTensor(std::string_view name)
+{
+    assert(name.size() <= sizeof(int64_t));
+    int64_t id = 0;
+    std::copy(name.begin(), name.end(), reinterpret_cast<char*>(&id));
```

Comment:
Formally speaking it must be a plain `std::memcpy` for such kind of tricks.

Reply:
`std::copy`  in C++ == `memcpy` in C for contiguous objects.

Reply:
Not exactly that, if we are talking about type punning.
`std::copy` is a typed operation and an stl algorithm - it's up to the library implementation to decide how to perform data copy.
Luckly we reinterpret it as a `char*`, that is probably safe.

Anyway, that is much safer:
```suggestion
    std::copy_n(name.begin(), sizeof(id), reinterpret_cast<char*>(&id));
```
In the worst case we truncate input name, but we will never cross dst buffer boundary and corrupt any data.

Reply:
yes, `copy_n` is a better idea.